onnxruntime/core/providers/coreml/builders/impl/squeeze_op_builder.cc

```diff
@@ -86,13 +86,15 @@ Status SqueezeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   if (model_builder.CreateMLProgram()) {
     using namespace CoreML::Specification::MILSpec;
 
-#if defined(TARGET_CPU_X86_64) && TARGET_CPU_X86_64
     // expand_dims has limited requirements for static shape, however, X86_64 has a bug that it can't handle scalar input
+    // in addition, MLProgram does not support scalar values -- we convert the scalars to 1D tensors.
```

Comment:
do we need to mention the x86_64 bug anymore if the ifdef is removed?

Reply:
Ah, I'm not sure what the x86_64 bug is and I thought to leave it in because it might refer to a separate issue from what I encountered, but I'm open to removing it

Reply:
@wejoncy would know more.

Reply:
CoreML will raise an internal error if feeding  a scalar input.
Not sure if the latest CoreML fixed it not.

Reply:
Gotcha looks to be the same issue as what I was running into so amended the comment